ultradns/resource_ultradns_dirpool.go

```diff
@@ -416,12 +429,36 @@ func populateResourceFromDirpool(d *schema.ResourceData, r *udnssdk.RRSet) error
 	} else {
 		d.Set("conflict_resolve", p.ConflictResolve)
 	}
-
+	log.Infof("r.RData= %v and p.RDataInfo= %v", r.RData, p.RDataInfo)
 	rd := makeSetFromDirpoolRdata(r.RData, p.RDataInfo)
 	err = d.Set("rdata", rd)
 	if err != nil {
 		return fmt.Errorf("rdata set failed: %v, from %#v", err, rd)
 	}
+
+	// Block created to set no_response
+
+	if r.Profile["noResponse"] != nil {
+		//noResponse := r.Profile["noResponse"].(map[string]interface{})
+		var noResponseBlock map[string]interface{}
+		np, err := makeDirpoolRdataInfoAPI(r.Profile["noResponse"])
+		if err != nil {
+			return fmt.Errorf("noResponseBlock unmarshalling failed: %v, from %#v", err, np)
+		}
+		var ts []map[string]interface{}
+		t := map[string]interface{}{
+			"all_non_configured": np.AllNonConfigured,
+			"geo_info":           mapFromGeoInfos(np.GeoInfo),
+			"ip_info":            mapFromIPInfos(np.IPInfo),
+		}
+		ts = append(ts, t)
+		err = d.Set("no_response", ts)
+		if err != nil {
+			return fmt.Errorf("noResponseBlock set failed: %v, from %#v", err, noResponseBlock)
```

Comment:
yup it can fail when the fields are not coming in from API as expected in case of DTO Upgrade so when the user will try to import it  it won't be importable in that case the import will fail and yup it won't make sense to the user in that case they will open a ticket and we will have to take a look into it